test/integration/marker_manager.cc

```diff
@@ -142,18 +143,29 @@ TEST_F(MarkerManagerTestFixture,
   // Show, but don't exec, so we don't block
   window->QuickWindow()->show();
 
-  // Check scene
-  auto engine = rendering::engine("ogre2");
-  ASSERT_NE(nullptr, engine);
+  bool receivedRenderEvent{false};
+  auto testHelper = std::make_unique<TestHelper>();
+  testHelper->forwardEvent = [&](QEvent *_event)
+  {
+    if (_event->type() == events::Render::kType)
+    {
+      receivedRenderEvent = true;
+    }
+  };
 
   int sleep = 0;
   int maxSleep = 30;
-  while (0 == engine->SceneCount() && sleep < maxSleep)
+  while (!receivedRenderEvent && sleep < maxSleep)
   {
     std::this_thread::sleep_for(std::chrono::milliseconds(100));
     QCoreApplication::processEvents();
     sleep++;
   }
+  EXPECT_TRUE(receivedRenderEvent);
+
+  // Check scene
+  auto engine = rendering::engine("ogre2");
+  ASSERT_NE(nullptr, engine);
```

Comment:
Interesting. Since this pattern is useful in most tests here and the order of calls is important, could this be refactored into a `testing::getRenderEngine` function that blocks until the window is available and calles `rendering::engine` at the right time?

Reply:
ok I added a helper function to do this. [acfcafc](https://github.com/gazebosim/gz-gui/pull/535/commits/acfcafc32af7024127e63add8c45eafe5e2e5731). One catch is that the function should only be called after the main window is visible. Added a doxygen comment on this.